python/foglamp/plugins/south/systeminfo/systeminfo.py

```diff
@@ -0,0 +1,366 @@
+                ifaces.append((iface_name, iface_addr))
+
+            return ifaces
+
+        def interface_transmission(dev, direction):
+            """Return the transmisson rate of a interface under linux
+            https://stackoverflow.com/a/41448191
+            """
+            path = "/sys/class/net/{}/statistics/{}_bytes".format(dev, direction)
+            f = open(path, "r")
+            bytes_collected = int(f.read())
+            f.close()
+            return bytes_collected
+
+        network_interfaces = get_all_network_interfaces()
+        network_traffic = {}
+        network_calc = {}
+        for interface_name, interface_ip in network_interfaces:
+            network_calc[interface_name] = {}
+            network_calc[interface_name]["bytes_recd_before"] = interface_transmission(interface_name, "rx")
+            network_calc[interface_name]["bytes_sent_before"] = interface_transmission(interface_name, "tx")
+
+        timestep = int(handle['networkSnifferPeriod']['value'])  # seconds
+        time.sleep(timestep)
+
+        for interface_name, interface_ip in network_interfaces:
+            network_calc[interface_name]["bytes_recd_after"] = interface_transmission(interface_name, "rx")
+            network_calc[interface_name]["bytes_sent_after"] = interface_transmission(interface_name, "tx")
+
+        for interface_name, interface_ip in network_interfaces:
+            network_traffic.update({
+                interface_name: {
+                    "IP": interface_ip,
+                    "networkPacketsReceived": network_calc[interface_name]["bytes_recd_after"] -
+                                              network_calc[interface_name]["bytes_recd_before"],
+                    "networkPacketsSent": network_calc[interface_name]["bytes_sent_after"] -
+                                          network_calc[interface_name]["bytes_sent_before"],
+                }
+            })
+        return network_traffic
+
+    def get_subprocess_result(cmd):
+        a = subprocess.Popen(cmd, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        outs, errs = a.communicate()
+        if a.returncode != 0:
+            raise OSError(
+                'Error in executing command "{}". Error: {}'.format(cmd, errs.decode('utf-8').replace('\n', '')))
+        d = [b for b in outs.decode('utf-8').split('\n') if b != '']
+        return d
+
+    def get_system_info():
+        data = {}
+        hostname = get_subprocess_result(cmd='hostname')[0]
+        data.update({
+            "hostName": hostname
+        })
+
+        # Generic search a list for an item starting with the given string. Return item stripped of
+        # the search string if success else return None
+        list_search = lambda x, xlist: next((y[len(x):] for y in xlist if y.startswith(x)), None)
+
+        # Get uptime, load, tasks, cpu usage, memory, swap memory information
+        c2 = get_subprocess_result(cmd='top -n1 -b')[:5]
```

Comment:
using top is not a good idea since top is an expensive command and utilise system resources. Moreover, the result will include the resources consumed by top itself.